src/forms/inputs/file-input/file-input.js

```diff
@@ -1,34 +1,45 @@
- * The value of this input is the data URL of the loaded file.
+ * The value of this input is a file object or an array of file objects with the `url` set to the base64 encoded data URL of the loaded file(s).
+ *
+ * Allowing multiple files to be selected requires passing in the `multiple` prop set to `true`. Multiple files can then be uploaded either all at once or piecemeal. This is different than the standard behavior of a file input, which will _replace_ any existing files with whatever is selected.
  *
- * An optional callback can be fired when the file is loaded: `onLoad(fileData, file)`.
- * This callback will be passed the data URL of the file, as well as the `File` object itself.
+ * Once a file has been read successfully, it is possible to remove the file object from the current set of values. An optional callback can be fired when a file is removed: `onRemove(removedFile)`. To customize the component that receives this `onRemove` handler, pass in a cutom component to the `removeComponent` prop.
  *
- * By default, this component displays a thumbnail preview of the loaded file. This preview can be customized
+ * By default, this component displays a thumbnail preview of the loaded file(s). This preview can be customized
  * by using the `thumbnail` or `hidePreview` props, as well as by passing a custom preview via `previewComponent` or `children`.
  *
  * A component passed using `previewComponent` will receive the following props:
- * - `file`: the uploaded file object, or `null` if no file has been uploaded.
- * - `value`: the current value of the input (data URL or empty string)
+ * - `value`: the current value of the input (file object or array of file objects)
  *
  * @name FileInput
  * @type Function
  * @param {Object} input - A `redux-forms` [input](http://redux-form.com/6.5.0/docs/api/Field.md/#input-props) object
  * @param {Object} meta - A `redux-forms` [meta](http://redux-form.com/6.5.0/docs/api/Field.md/#meta-props) object
- * @param {Function} [onLoad] - A callback fired when the file is loaded
+ * @param {Function} [readFiles] - A callback that is fired with new files and is expected to return an array of file objects with the `url` key set to the "read" value. This can be either a data URL or the public URL from a 3rd party API
+ * @param {Boolean} [multiple=false] - A flag indicating whether or not to accept multiple files
+ * @param {Function} [onRemove=noop] - A callback fired when the file is removed (only available when `multiple` is set to `true`)
+ * @param {Function} [removeComponent=RemoveButton] - A custom component that receives the `onRemove` callback (only available when `multiple` is set to `true`)
  * @param {String} [thumbnail] - A placeholder image to display before the file is loaded
- * @param {Boolean} [hidePreview] - A flag indicating whether or not to hide the file preview
+ * @param {Boolean} [hidePreview=false] - A flag indicating whether or not to hide the file preview
```

Comment:
Should `previewComponent` be listed here as a prop option?

Reply:
I also just had a chat with Mike who wasn't aware that you could pass `accept` with a list of the only accepted filetypes. I know that's a default `input` prop, but could be helpful to make everyone aware it's available with this input.